builder/azure/arm/builder.go

```diff
@@ -166,31 +166,27 @@ func (b *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packe
 	deploymentName := b.stateBag.Get(constants.ArmDeploymentName).(string)
 
 	if b.config.OSType == constants.Target_Linux {
-		steps = []multistep.Step{
-			NewStepCreateResourceGroup(azureClient, ui),
-			NewStepValidateTemplate(azureClient, ui, b.config, GetVirtualMachineDeployment),
-			NewStepDeployTemplate(azureClient, ui, b.config, deploymentName, GetVirtualMachineDeployment),
-			NewStepGetIPAddress(azureClient, ui, endpointConnectType),
-			&communicator.StepConnectSSH{
-				Config:    &b.config.Comm,
-				Host:      lin.SSHHost,
-				SSHConfig: b.config.Comm.SSHConfigFunc(),
-			},
-			&packerCommon.StepProvision{},
-			&packerCommon.StepCleanupTempKeys{
-				Comm: &b.config.Comm,
-			},
-			NewStepGetOSDisk(azureClient, ui),
-			NewStepGetAdditionalDisks(azureClient, ui),
-			NewStepPowerOffCompute(azureClient, ui),
-			NewStepCaptureImage(azureClient, ui),
-			NewStepDeleteResourceGroup(azureClient, ui),
-			NewStepDeleteOSDisk(azureClient, ui),
-			NewStepDeleteAdditionalDisks(azureClient, ui),
-		}
+		steps = append(steps,
```

Comment:
Steps needs to be initialized first, so the first time you can just do:
```suggestion
		steps = []multistep.Step{
```